### ISSUE TITLE
[Feature Request] Add option to print results to console

### ISSUE THREAD

**Issue by logan-markewich** [AIR-Bench/AIR-Bench#26]
From what I can tell, the actual eval metrics are never printed to console (at least with a custom retriever). 

It would be valuable to have a way to print these, either during or after running the benchmark

**Comment by hanhainebula** [comment]
Hi, @logan-markewich! I am not sure if I have correctly understood your meaning. 

We did not set a fixed metric for each dataset like MTEB did. Actually, on the user side, users only need to generate top-k search results and then submit the search results to our leaderboard. We compute metrics such as ndcg@10, recall@100, etc. in our backend machine and then upload the eval results to the leaderboard. You can use the metric selector to select the metric you want to use.

Thank you.

**Comment by logan-markewich** [comment]
@hanhainebula thats right. But then, I don't know what the eval results will be until it's uploaded and published to the leaderboard

Offering some utils to calculate the metrics used on the leaderboard would be nice. Then users aren't uploading results blindly and hoping the scores are ok

**Comment by hanhainebula** [comment]
We use the [code](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/evaluation/evaluators/RetrievalEvaluator.py#L467) in MTEB to evaluate using the saved search results.

> But then, I don't know what the eval results will be until it's uploaded and published to the leaderboard.
 
About this, we have considered it during the development phase. But we didn't come up with a good strategy to achieve this. Therefore, we designed an anonymous submission mechanism that allows users to submit anonymously. However, this mechanism is not an ideal solution. If you have any ideas, please let us know. Thank you!

**Comment by logan-markewich** [comment]
I think just even running that code on the results and printing it after each dataset runs would be a huge improvement

In the meantime, thanks for pointing me out to the code. I'll run this manually for now on my own results (but you can imagine that if this ran automatically after each dataset runs in air-bench, it would be a nice UX improvement I think)

**Comment by nan-wang** [comment]
@logan-markewich Do you think offering a DEV set could solve your concern? I see there is a dilemma. On one side, we want to do a close test. On the other side, the developers don't want to submit blindly. I guess having a dev set could solve the issue.

**Comment by lulu51230** [comment]
Developers may need to calculate the code for the ndcg@10 indicator or other indicators to determine whether they want to submit the list.
